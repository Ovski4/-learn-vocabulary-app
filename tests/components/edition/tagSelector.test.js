import reducer from '../../../src/reducers/index';
import { createStore } from 'redux';
import React from 'react';
import TagSelector from '../../../src/components/edition/tagSelector/connect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'

describe('TagSelector component', () => {

    it('Should render the component with the english locale', () => {

        const store = createStore(reducer, {
            tags: [
                {
                    id: '1',
                    label: 'test'
                },
                {
                    id: '2',
                    label: 'test2'
                }
            ],
            config: {
                locale: 'en'
            }
        });

        const rendered = renderer.create(
            <Provider store={store}>
                <TagSelector initialSelectedTags={[{
                    id: '1',
                    label: 'test'
                }]}/>
            </Provider>
        ).toJSON();
        expect(rendered).toMatchSnapshot();
    });

});
