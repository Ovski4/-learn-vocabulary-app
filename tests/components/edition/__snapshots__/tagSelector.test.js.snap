// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagSelector component Should render the component with the english locale 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#eeeeee",
        "borderRadius": 2,
        "marginLeft": 5,
      }
    }
  >
    <View
      style={
        Object {
          "height": 32,
        }
      }
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "Pick a tag",
              "textColor": undefined,
              "value": null,
            },
            Object {
              "label": "test",
              "textColor": undefined,
              "value": "1",
            },
            Object {
              "label": "test2",
              "textColor": undefined,
              "value": "2",
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  />
</View>
`;
